sml_builder/__init__.py

```diff
@@ -84,3 +85,45 @@ def set_variables():
     return {
         "docs_integration_active": docs_integration["enabled"],
     }
+
+
+@app.before_request
+def build_help_centre_structure():
+    category_urls = {
+        "Information": "information",
+        "Access (and usage)": "access",
+        "Feedback": "feedback",
+        "Support": "support",
+    }
+    app.before_request_funcs[None].remove(build_help_centre_structure)
+    nav = {"categories": []}
+    contents = getContent("helpCentreInformation")
+    submit_request = getContent("helpCentreMethodRequest")
+    unique = set(d["help_centre_category"] for d in contents)
+    for category in unique:
+        nav["categories"].append(
+            {"name": category_urls[category], "label": category, "subcategories": []}
+        )
+        for content in contents:
+            if content["help_centre_category"] == category:
+                nav["categories"][-1]["subcategories"].append(
+                    {"name": content["id"], "label": content["title"]}
+                )
+            if not checkEmptyList(submit_request):
+                if category == "Information":
+                    if not any(
+                        d["name"] == "methods-request"
+                        for d in nav["categories"][-1]["subcategories"]
+                    ):
+                        nav["categories"][-1]["subcategories"].append(
+                            {
+                                "name": "methods-request",
+                                "label": "Submit a method request",
+                            }
+                        )
+    nav["categories"] = sorted(nav["categories"], key=lambda x: x["name"])
+    for category in nav["categories"]:
+        category["subcategories"] = sorted(
+            category["subcategories"], key=lambda x: x["label"]
+        )
+    app.cache["help_centre_nav"] = nav
```

Comment:
cache is a dictionary object within the flask application object, each time we deploy that application will be recreated, so in the case of this method we deploy, and during the first request to it we create this navigation and add it to the cache dictionary, it will remain here unless deleted or the app is stopped